### DISCUSSION TITLE
Shell expansion on the command line

### DISCUSSION THREAD

**Discussion by OJ** [neovim/neovim#29654] [Q&A]
Hi Folks,

Long term OG Vi/Vim user here. Forgive me for asking a potentially silly question. But this little difference between Vim and NeoVim is driving me a bit nuts.

It's not uncommon for me to load a bunch of files at the same time from the CLI:
```
C:\thing> vim *
```
In this case, Vim opens a single editor instance with all files in the current folder open in different buffers. This also works with gVim.

If I do the same thing with NeoVim, it refuses to work. NeoVim shows an error:
```
E303: Unable to open swap file for "*", recovery impossible
```
It appears that it's not doing shell expansion from the command line, and instead trying to open a file called `*`.

Is this a bug in the application or is there a setting that I'm missing somewhere that'll make this work?

Thanks for the help.

**Comment by zeertzjq** [comment]
I'm not sure if this should be supported, but you can probably use `nvim +n*` instead.

**Reply by OJ** [reply]
Thanks for the response. This doesn't do what I'm looking for. Your suggestion includes folders, not just files.

What I'm looking for is the same support that you get with almost every other editor: the ability to specify multiple, specific files on the command line with wildcard expansion without the editor trying to open folders.

Thanks again for your time.

**Reply by zeertzjq** [reply]
Hmm, it's a bit surprising that "every other editor" does this, because in most shells (Bash, Zsh, Fish, etc.) the expansion of `*` includes folders.

**Reply by OJ** [reply]
Aye, true ;)

**Reply by OJ** [reply]
My use of "every other editor" might be a bit over the top. Vim clearly does the smarts here to exclude folders and just open files (at least on my test boxes of Windows 11 and Fedora).

I just want my `vim *.foo` back!

**Reply by zeertzjq** [reply]
Hmm, what shells are you using on these platforms?
